src/configurationProvider.ts

```diff
@@ -225,17 +306,24 @@ export class JavaDebugConfigurationProvider implements vscode.DebugConfiguration
         }
 
         const pickItems: IMainClassQuickPickItem[] = this.formatRecentlyUsedMainClassOptions(res);
+        const selected = await this.showMainClassQuickPick(pickItems, "Select main class<project name>");
+        if (selected) {
+            this.debugHistory.updateMRUTimestamp(selected);
+        }
 
-        const selection = pickItems.length > 1 ?
-            await vscode.window.showQuickPick(pickItems, { placeHolder: "Select main class<project name>" })
-            : pickItems[0];
+        return selected;
+    }
 
-        if (selection && selection.item) {
-            this.debugHistory.updateMRUTimestamp(selection.item);
-            return selection.item;
-        } else {
-            return undefined;
+    private async showMainClassQuickPick(pickItems: IMainClassQuickPickItem[], quickPickHintMessage: string, autoPick: boolean = true):
+        Promise<IMainClassOption | undefined> {
+        if (pickItems.length === 0) {
+            return;
```

Comment:
return nothing means resolving with undefined. This does not seem right. I would recommend through an exception because an empty candidate list is an error case.

Reply:
will remove the empty check logic, because the check has been done in the caller side. No need to duplicate it.